### PR TITLE
Add SentinelOne for Linux support 

### PR DESCRIPTION
**Pull request by MarcSamD**

### CODE REVIEW

File: lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+sub _getSentineloneInfo {
+	my @command1 = getAllLines(command => '/opt/sentinelone/bin/sentinelctl version');
+	my @command2 = getAllLines(command => '/opt/sentinelone/bin/sentinelctl engines status');
+	my @command3 = getAllLines(command => '/opt/sentinelone/bin/sentinelctl control status');
+	my @command4 = getAllLines(command => '/opt/sentinelone/bin/sentinelctl management status');
+
+    my @output = (@command1,@command2,@command3,@command4)
+        or return;
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm]: I see here one problem and a cleaner way to prepare `@output`. The problem is not critical. You just forgot to get `%params` and include it in `getAllLines()` calls so the run command is logged. Also it's better to return on first error and `push` gotten lines into the still declared `@output` array as we got them.

It seems also you're using tabulation for indentation. Please, can you reformat your source to use 4 spaces indentation ?

```suggestion
sub _getSentineloneInfo {
    my (%params) = @_;

    my @output = getAllLines(command => '/opt/sentinelone/bin/sentinelctl version', %params)
        or return;
    push @output, getAllLines(command => '/opt/sentinelone/bin/sentinelctl engines status', %params)
        or return;
    push @output, getAllLines(command => '/opt/sentinelone/bin/sentinelctl control status', %params)
        or return;
    push @output, getAllLines(command => '/opt/sentinelone/bin/sentinelctl management status', %params)
        or return;
```

File: lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    my $av = {
+        NAME     => 'SentinelAgent',
+        COMPANY  => 'SentinelOne',
+        ENABLED => 0,
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm]: You should keep same indentation on `=>` as you want to keep values aligned.
```suggestion
        ENABLED  => 0,
```

File: lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm (hunk 1 above)
Comment (g-bougard) [reply lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm]: Indeed I've another suggestion:
```suggestion
sub _getSentineloneInfo {
    my (%params) = @_;

    my $cmd = '/opt/sentinelone/bin/sentinelctl';

    my @output = getAllLines(
        command => "$cmd version && $cmd engines status && $cmd control status && $cmd management status",
        %params
    )
        or return;
```
Can you try this one is giving the expected result ? It should and it looks indeed better.

File: lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        } elsif ($key eq "DFI library version") {
+            $av->{BASE_VERSION} = $value;
+        } elsif ($key eq "Agent state") {
+            $av->{ENABLED} = $value eq "Enabled" ? 1 : 0;
+        } elsif ($key eq "Connectivity") {
+            # SentinelAgent does not directly report "uptodate" status but we can assume it is updated if the cloud connectivity is working.
+            $av->{UPTODATE} = $value eq "On" ? 1 : 0;
+        # Not this item as SentinelOne antivirus is not using the traditional signatures definition model
+		# } elsif ($key eq "Last security content update" && $value =~ /^(\d{4}-\d+-\d+) at/) {
+        #    $av->{BASE_CREATION} = $1;
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm]: You can just remove this comment. Also you have tabulations for indentation on l.63. If you want to keep it, replace the tabulations by spaces.

**Review by g-bougard — CHANGES_REQUESTED** [review]
Hello @MarcSamD 

thank you for your pull request.

I see a little improvement to achieve on the `_getSentineloneInfo()` API definition. Better use my second comment there if it's okay for you.

I saw you mixed tabulations and spaces for indentation. This is not a big deal with perl but keep using spaces for this project.

Finally, have you some output samples I can use to create a unittest ?

File: lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm (hunk 3 above)
Comment (MarcSamD) [reply lib/GLPI/Agent/Task/Inventory/Linux/AntiVirus/Sentinelone.pm]: Comments removed. Not sure for the indentation though.

**Comment by MarcSamD** [comment]
Hi @g-bougard ,
Thank you for looking at this PR. I have very little knowledge in development so forgive me for submitting such a messy code.

Below output sample of the 4 commands:

```
Agent version: 30.1.1.10
SentinelCTL version: 30.1.1.10
Ranger version: 30.5.5.9
Git hash: 123456789abcde123456789abcde
DFI engine                          Enabled
DFI library                         Initialized
DFI library version                 30.5.6.5
DFI library git hash                d3dfc2a1641635f8e319257ab2314887fd0bc62d-Release.x64
Dynamic engine                      Enabled
Application control (container)     Disabled
Application control (host)          Disabled
Agent state      Enabled

Process Name     PID
orchestrator     901
network          902
scanner          903
agent            904
firewall         905
Anti Tamper                      On
AppInventory force update        86400
AppInventory interval            600
Connect through                  direct
Connectivity                     On
Customer id                      undefined
Device type                      server
Machine information interval     600
Override DV proxy address        undefined
Proxy address                    undefined
Secured protocol mode            best_effort
Site-Key                         12345abcdef6789
URL                              https://server-101.sentinelone.net
UUID                             123a123-12d12-1234-ab0c-1234bd1234
```

**Comment by g-bougard** [comment]
Hi @MarcSamD 

don't worry about your coding level, we are here to help finalize stuffs. Contributors with nice ideas will always be welcome.

But I still thank you for this nice still working PR, and thank you for the sample output sharing, I'll be able to add a test with it.

**Pull request merged**